[pythonscripting] Clean fails on Windows platform

I use Windows git-bash to do an add-on build with `mvn clean install` (and a few more options).
Build fails with

`
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:3.5.0:clean (default-clean) on project org.openhab.automation.pythonscripting: Failed to clean project: Failed to delete bundles\org.openhab.automation.pythonscripting\target\python\.git\objects\pack\pack-....idx -> [Help 1]
`

I cannot do a make clean on the checked out git repo.
Manually removing the folder works:
`
❯ rm -rf bundles/org.openhab.automation.pythonscripting/target
`

Not sure which background process is locking those files.


<!-- Provide a general summary of the issue in the *Title* above -->
<!-- If the issue is related to a binding, please include its short name in -->
<!-- square brackets in the title - Example: "[astro] My issue..." -->

<!-- Important: Please contact the openHAB community forum for questions or -->
<!-- for configuration and usage guidance: https://community.openhab.org -->

<!-- Feel free to delete any comment lines in the template (starting with "<!--") -->

## Expected Behavior
<!-- If you're describing a bug, tell us what should happen -->
<!-- If you're suggesting a change/improvement, tell us how it should work -->
build succeeds without error

## Current Behavior
<!-- If describing a bug, tell us what happens instead of the expected behavior -->
<!-- Include related log information (preferably debug level) and related configs -->
<!-- Use a file attachment for log and config information longer than a few lines -->
<!-- Enclose multi-line log/code snippets with ``` on new lines for proper formatting -->
<!-- If suggesting a change/improvement, explain the difference from current behavior -->
<!-- For improvements, discuss at community.openhab.org first and include link to topic -->

manual interaction necessary

## Possible Solution
<!-- Not obligatory, but suggest a fix/reason for the bug, -->
<!-- or ideas how to implement the addition or change -->

I checked if we could do a sparse checkout or checkout without history.
Or mvn git plugin is restriced and does not support those special commands (tbc, maybe I am wrong).

A second way would be to configure the `maven-clean-plugin` via the pom file (just for this binding).
```
      <plugin>
        <artifactId>maven-clean-plugin</artifactId>
        <version>3.5.0</version>
        <configuration>
          <failOnError>false</failOnError>
        </configuration>
      </plugin>
```
This works, but issues ugly warnings.

## Steps to Reproduce (for Bugs)
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->

build, and then clean

1. mvn install
2. mvn clean install

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version used: (e.g., openHAB and add-on versions)
* Environment name and version (e.g. Chrome 76, Java 8, Node.js 12.9, ...):
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...):
Windows 11, git bash, zsh

--- a/bundles/org.openhab.automation.pythonscripting/pom.xml
+++ b/bundles/org.openhab.automation.pythonscripting/pom.xml
@@ -93,6 +93,13 @@
           <pmdFilter>${project.basedir}/suppressions.properties</pmdFilter>
         </configuration>
       </plugin>
+      <!-- workaround for failing clean of checked out git repo breaking the Windows build, #19250 -->
+      <plugin>
+        <artifactId>maven-clean-plugin</artifactId>
+        <configuration>
+          <force>true</force>
+        </configuration>
+      </plugin>
     </plugins>
   </build>
 </project>

